pkg/controller/jaeger/upgrade.go

```diff
@@ -22,7 +22,12 @@ func (r *ReconcileJaeger) applyUpgrades(ctx context.Context, jaeger v1.Jaeger) (
 		if jaeger.Status.Version != currentVersions.Jaeger {
 			// in theory, the version from the Status could be higher than currentVersions.Jaeger, but we let the upgrade routine
 			// check/handle it
-			upgraded, err := upgrade.ManagedInstance(ctx, r.client, jaeger, currentVersions.Jaeger)
+			versions, err := upgrade.Versions()
```

Comment:
I don't think the `Versions` should be exposed in a public method. I like the fact that the parsing of the versions is now only done once and cached, so that aspect of the changes should remain.

The main reason this change was done was because the parsing of the version was used as an error to demonstrate skipping the upgrade for a jaeger instance. This type of error is actually very unlikely.

Therefore, if you could make the versions function package protected, but for the skip test instead create an upgrade function that returns an error?